### ISSUE TITLE
Adding new api to whitelist

### ISSUE THREAD

**Issue by KamranMiadi** [RamyElkest/libvirt-node#43]
Hi.
Im using libvirt-node in my project and i needed some api for monitoring my VMs.
i added them to whitelist as you said in DEVELOPMENT.md but when I try to regenerate the files I get the following error:

```
Error: Type 'virDomainPtr *' is not mapped to a napi type
```
I see that you mentioned this error in whitelist.js.
Is there anyway to fix it?
I would be thankful if you consider helping me .
@RamyElkest

**Comment by RamyElkest** [comment]
Hi @KamranMiadi,

Thanks for using this library.

We need to add support pointer returns (aka lists).
This would actually unlock a few of the whitelists.

in impl_default.js we need to customise the return to check if it's an array then build the array and return it (currently we only handle scalar returns with exception of impl_list_all which we could use as an example, as well as https://github.com/libvirt/libvirt-python/blob/master/libvirt-override-virConnect.py#L664)

I can have a look at it this weekend if that's works for you,
otherwise if you're interested in giving it a go -- I'd be more than happy to help 👍

**Comment by KamranMiadi** [comment]
Thanks for your answer @RamyElkest 
I will put more time to it and inform you if I made any progress.
Thanks again. :pray:

**Comment by KamranMiadi** [comment]
Hi @RamyElkest .
Can you look it up this weekend?
I tried myself but couldn't fix it.

**Comment by RamyElkest** [comment]
Yup -- giving it a go

@KamranMiadi can you confirm which method(s) you're interested in/missing?

I presume `virDomainListGetStats` but any more?

**Comment by KamranMiadi** [comment]
yeah i need that and i need functions to add gpu to VM and monitor gpu as swell
but haven't checked which functions they are.
@RamyElkest

**Comment by KamranMiadi** [comment]
Hi @RamyElkest .
Can you be more specific on how to fix this issue? 
I've been trying to fix it with what you said in earlier comments for more than a week. but had no luck.
I can't figure out how impl_default.js works.
Also functions `listAllStoragePools()` and  `listAllDomains()` return error 
`Assertion status == napi_ok' failed.`.
I guess they return list too.
